rest/model/admin.go

```diff
@@ -627,6 +629,59 @@ func (a *APIJiraConfig) ToService() (interface{}, error) {
 	}, nil
 }
 
+type APILDAPRoleMapping struct {
+	LDAPGroup APIString `json:"ldap_group"`
+	RoleID    APIString ` json:"role_id"`
+}
+
+func (a *APILDAPRoleMapping) BuildFromService(h interface{}) error {
+	switch v := h.(type) {
+	case evergreen.LDAPRoleMapping:
+		a.LDAPGroup = ToAPIString(v.LDAPGroup)
+		a.RoleID = ToAPIString(v.RoleID)
+	}
+
+	return nil
+}
+
+func (a *APILDAPRoleMapping) ToService() (interface{}, error) {
+	mapping := evergreen.LDAPRoleMapping{
+		LDAPGroup: FromAPIString(a.LDAPGroup),
+		RoleID:    FromAPIString(a.RoleID),
+	}
+
+	return mapping, nil
+}
+
+type APILDAPRoleMap []*APILDAPRoleMapping
```

Comment:
Could these be structs and not pointers?

Reply:
Maybe? I am following the pattern of other substructs, they are all pointers

Reply:
Those are pointers to allow null values as the field itself. It doesn't make much sense to have null values within an array